upb/json/parser.rl

```diff
@@ -2002,6 +2012,7 @@ static bool start_subobject(upb_json_parser *p) {
     inner->m = upb_fielddef_msgsubdef(p->top->f);
     set_name_table(p, inner);
     inner->f = NULL;
+    inner->is_repeated = false;
```

Comment:
Can you factor out the code that sets the inner frame to default values?  Something like:

inner = init_frame(p);